cmd/juju/cloud/updatepublicclouds.go

```diff
@@ -125,16 +137,19 @@ func (c *updatePublicCloudsCommand) Run(ctxt *cmd.Context) error {
 		return errors.Trace(err)
 	}
 	fmt.Fprint(ctxt.Stderr, "Fetching latest public cloud list...\n")
-	publishedClouds, err := PublishedPublicClouds(c.publicCloudURL, c.publicSigningKey)
-	if err != nil {
-		return errors.Trace(err)
-	}
 	var returnedErr error
-	if c.Client {
-		if err := c.updateClientCopy(ctxt, publishedClouds); err != nil {
-			ctxt.Infof("ERROR %v", err)
-			returnedErr = cmd.ErrSilent
+	publishedClouds, msg, err := FetchAndMaybeUpdatePublicClouds(
+		PublicCloudsAccessDetails{c.publicSigningKey, c.publicCloudURL},
+		c.Client)
+	if err != nil {
+		if len(publishedClouds) == 0 {
```

Comment:
I'm unsure about this, as normally we always `return nil` for the published clouds, although in this instance I'm actually unsure the path forward. Maybe add a comment stating the fact?

Reply:
Added comment. Essentially since the Fetch may also update client copy, the error may be either with the retrieval of the public clouds or with the client copy update. If we got an error and no clouds, we know that the error is with the retrieval which means that there is no point trying to update a controller.